### PR TITLE
[merged] Add support for ostree static-delta delete

### PR DESCRIPTION
**Pull request by alexlarsson**

### CODE REVIEW

File: src/ostree/ot-builtin-static-delta.c [hunk 1]
```diff
@@ -169,6 +171,39 @@ ot_static_delta_builtin_show (int argc, char **argv, GCancellable *cancellable,
...
 static gboolean
+ot_static_delta_builtin_delete (int argc, char **argv, GCancellable *cancellable, GError **error)
+{
+  gboolean ret = FALSE;
+  GOptionContext *context;
+  glnx_unref_object OstreeRepo *repo = NULL;
+  const char *delta_id = NULL;
+
+  context = g_option_context_new ("SHOW - Dump information on a delta");
+
```
Comment (giuseppe) [inline src/ostree/ot-builtin-static-delta.c]: this should be changed to refer to DELETE

File: src/libostree/ostree-repo-static-delta-core.c [hunk 2]
```diff
@@ -783,6 +783,29 @@ _ostree_delta_needs_byteswap (GVariant *superblock)
...
+{
+  gboolean ret = FALSE;
+  g_autofree char *from = NULL;
+  g_autofree char *to = NULL;
+  g_autofree char *deltadir = NULL;
+
+  _ostree_parse_delta_name (delta_id, &from, &to);
+  deltadir = _ostree_get_relative_static_delta_path (from, to, NULL);
+
+  if (!glnx_shutil_rm_rf_at (self->repo_dir_fd, deltadir,
```
Comment (giuseppe) [inline src/libostree/ostree-repo-static-delta-core.c]: `glnx_shutil_rm_rf_at` does not return an error on ENOENT.  Should we return an error if the delta directory does not exist?

**Comment by giuseppe** [comment]
It misses the documentation in `man/ostree-static-delta.xml`

**Comment by alexlarsson** [comment]
fixed. Also added "show" to the docs

**Comment by giuseppe** [comment]
thanks.  LGTM, waiting for @cgwalters if he agrees on the new CLI interface

File: tests/test-delta.sh [hunk 3]
```diff
@@ -186,3 +186,18 @@ ${CMD_PREFIX} ostree --repo=repo2 fsck
 ${CMD_PREFIX} ostree --repo=repo2 ls ${newrev} >/dev/null
 
 echo 'ok apply offline inline'
+
+${CMD_PREFIX} ostree --repo=repo static-delta list | grep ^${origrev}-${newrev}$ || exit 1
+${CMD_PREFIX} ostree --repo=repo static-delta list | grep ^${origrev}$ || exit 1
```
Comment (cgwalters) [inline tests/test-delta.sh]: I'm confused by `|| exit 1`, don't we get that by our use of `-o pipefail`?

**Comment by cgwalters** [comment]
No objections here, @giuseppe want to add the r+?

**Comment by giuseppe** [comment]
sure, looks fine for me:

@cgwalters-bot r+ c385dcc

File: tests/test-delta.sh (hunk 3 above)
Comment (alexlarsson) [inline tests/test-delta.sh]: Quite possible. I just copied those lines from earlier in the same file.

**Pull request closed without merging**